src/libpgagroal/management.c

```diff
@@ -1236,42 +1258,52 @@ pgagroal_management_read_isalive(SSL* ssl, int socket, int* status, char output_
 
    *status = pgagroal_read_int32(&buf[MANAGEMENT_INFO_SIZE]);
 
-   // do I need to provide JSON output?
-   if (output_format == COMMAND_OUTPUT_FORMAT_JSON)
-   {
-      char* version_buf = NULL;
 
-      if (pgagroal_executable_version_string(&version_buf, version))
-      {
-         goto error;
-      }
+   char* version_buf = NULL;
 
-      cJSON* json = pgagroal_json_create_new_command_object("ping", true, pgagroal_executable_name(application), version_buf);
-      cJSON* output = pgagroal_json_extract_command_output_object(json);
+   if (pgagroal_executable_version_string(&version_buf, version))
+   {
+      goto error;
+   }
 
-      cJSON_AddNumberToObject(output, "status", *status);
+   json = pgagroal_json_create_new_command_object("ping", true, pgagroal_executable_name(application), version_buf);
+   output = pgagroal_json_extract_command_output_object(json);
 
-      if (*status == PING_STATUS_RUNNING)
-      {
-         cJSON_AddStringToObject(output, "message", "running");
-      }
-      else if (*status == PING_STATUS_SHUTDOWN_GRACEFULLY)
-      {
-         cJSON_AddStringToObject(output, "message", "shutdown gracefully");
-      }
-      else
-      {
-         cJSON_AddStringToObject(output, "message", "unknown");
-      }
-
-      return pgagroal_json_print_and_free_json_object(json);
+   pgagroal_json_put(output, "status", (uintptr_t)  *status, ValueInt32);
 
+   if (*status == PING_STATUS_RUNNING)
+   {
+      pgagroal_json_put(output, "message", (uintptr_t)  "running", ValueString);
+   }
+   else if (*status == PING_STATUS_SHUTDOWN_GRACEFULLY)
+   {
+      pgagroal_json_put(output, "message", (uintptr_t)  "shutdown gracefully", ValueString);
+   }
+   else
+   {
+      pgagroal_json_put(output, "message", (uintptr_t)  "unknown", ValueString);
    }
 
+   // do I need to provide JSON output?
+   if (output_format == COMMAND_OUTPUT_FORMAT_JSON)
+   {
+      return pgagroal_print_and_free_object(json,FORMAT_JSON);
```

Comment:
I'm not sure this function should not have `json` in its name. While I agree that we could have more formats in the future, the function will always *print from a json object* and *free a json object*. The printing could be in YAML, as an example, but we start and free always a json struct, right?